Dockerfile.debian

```diff
@@ -59,221 +67,162 @@ RUN apt-get -y update && apt-get -y install \
-    curl -LO https://dl.minio.io/client/mc/release/linux-amd64/mc && \
-    chmod +x mc && mv mc /usr/local/bin
-
-# redis-benchmark, redis-cli
-RUN cd /tmp && \
-    curl -LO http://download.redis.io/releases/redis-4.0.10.tar.gz && \
-    mkdir -p redis && \
-    tar --strip-components=1 -C redis -xzf redis-4.0.10.tar.gz && \
-    make -C redis && \
-    cp redis/src/redis-cli redis/src/redis-benchmark /usr/local/bin &&  \
-    rm -rf redis
-
-# terraform
-RUN cd /tmp && \
-    curl -Lo terraform.zip https://releases.hashicorp.com/terraform/0.11.7/terraform_0.11.7_linux_amd64.zip && \
-    unzip terraform.zip && chmod +x terraform && mv terraform /usr/local/bin && rm -f terraform.zip
-
-# testssl
-RUN git clone --depth 1 https://github.com/drwetter/testssl.sh.git /usr/local/share/testssl.sh && \
-    ln -s /usr/local/share/testssl.sh/testssl.sh /usr/local/bin/testssl && chmod +x /usr/local/bin/testssl
-
-# tmux
-RUN cd /tmp && curl -L https://github.com/libevent/libevent/releases/download/release-2.0.22-stable/libevent-2.0.22-stable.tar.gz | tar xz && \
-    cd libevent-* && \
-    ./configure && make && sudo make install && \
-    cd .. && rm -fr libevent-* && \
-    cd /tmp && curl -L https://github.com/tmux/tmux/releases/download/2.7/tmux-2.7.tar.gz | tar xz && \
-    cd tmux-* && \
-    ./configure --prefix=/usr/local && \
-    make && sudo make install && \
-    cd .. && rm -fr tmux-*
-
-# vim
-RUN cd /tmp && curl -L https://github.com/vim/vim/archive/v8.1.0177.tar.gz | tar xz && \
-    cd vim-* && \
-    ./configure \
-        --with-features=huge \
-        --enable-cscope \
-        --enable-rubyinterp \
-        --enable-luainterp \
-        --enable-perlinterp \
-        --enable-pythoninterp \
-        --enable-python3interp \
-        --enable-tclinterp \
-        --enable-multibyte \
-        --prefix=/usr/local && \
-    make VIMRUNTIMEDIR=/usr/local/share/vim/vim80 && \
-    sudo make install && \
-    cd .. && rm -fr vim-*
-
-# yadm
-RUN curl -fLo /usr/local/bin/yadm https://github.com/TheLocehiliosan/yadm/raw/master/yadm && \
-    chmod a+x /usr/local/bin/yadm
-
-# zsh
-RUN cd /tmp && curl -L https://sourceforge.net/projects/zsh/files/zsh/5.5.1/zsh-5.5.1.tar.gz/download | tar xz && \
-    cd zsh-* && \
-    ./configure --with-tcsetpgrp --prefix=/usr/local && make && sudo make install && \
-    sudo echo "/usr/local/bin/zsh" >> /etc/shells && \
-    cd .. && rm -fr zsh-*
-
+# We sometimes install or build from source to get specific versions.
+
+RUN echo "Install/build apps that are not provided by the OS packages." && \
+    # git (needs to build first as a dependency) && \
+        cd /tmp && curl -sSL https://github.com/git/git/archive/v2.18.0.tar.gz | tar xz && \
+        cd git-* && make configure && ./configure --prefix=/usr/local && make && make install && \
+        cd .. && rm -fr git-* && \
+    # dep && \
+        curl -sSLo dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64 && \
```

Comment:
Build args might be neat for versions.
https://docs.docker.com/engine/reference/builder/#arg